generate/generate.js

```diff
@@ -0,0 +1,253 @@
+const https = require("https");
+const fs = require("fs");
+
+// Get slug as argument
+const slug = process.argv[2];
+
+// Download the canonical data
+let dataUrl = `https://raw.githubusercontent.com/exercism/problem-specifications/main/exercises/${slug}/canonical-data.json`;
+https
+  .get(dataUrl, (res) => {
+    let body = "";
+    res.on("data", (chunk) => (body += chunk));
+    res.on("end", () => {
+      try {
+        let canonicalData = JSON.parse(body);
+        generateFiles(slug, canonicalData);
+      } catch (error) {
+        console.error(error.message);
+      }
+    });
+  })
+  .on("error", (error) => {
+    console.error(error.message);
+  });
+
+// Use the canonical data to generate the multiple files
+function generateFiles(slug, canonicalData) {
+  const exercise = kebabToPascal(slug);
+
+  let extractedFunctions = {};
+  extractFunctions(canonicalData, extractedFunctions);
+
+  const functionList = Object.entries(extractedFunctions)
+    .map(([key, _]) => key)
+    .join(", ");
+
+  const allFunctionsCode = Object.entries(extractedFunctions)
+    .map(([key, value]) => generateFunctionCode(key, value))
+    .join("\n\n");
+
+  const mainFile = `
+module ${exercise} exposing (${functionList})
+
+${allFunctionsCode}
+`;
+  console.log(mainFile);
+
+  const allTestCode = generateAllTestsCode(
+    exercise,
+    extractedFunctions,
+    canonicalData
+  );
+
+  const testFile = `
+module Tests exposing (tests)
+
+import ${exercise}
+import Expect
+import Test exposing (Test, describe, skip, test)
+
+${toElmComments(canonicalData.comments)}
+
+tests : Test
+tests = describe "${exercise}" [ ${allTestCode} ]
```

Comment:
One thing that's missing is removing the first `skip <|` in the test file.

Reply:
Yes, adding all the skips is not something that I found useful when I was adding an exercise. I wanted to discuss this with you by the way. In the template, I've added a commented skip `-- skip <|` so that people can run `elm-test-rs` directly in the exercise directory.

My workflow was to first develop the solution in the `src/Main.elm`, run the tests while writing the solution. And when the solution is done, copy it to the example and remove the implementatiin from the template.

Considering this workflow, it was easier to be able to run the tests directly. Then when I've finished, uncomment all the `-- skip` except for the first one.

Is your workflow different ? Do you run `bin/build.sh` instead of directly run the tests of just the current exercise yourself?

Reply:
Oh, I see, that's rather clever. I guess I also uncomment the skips first, and add them back later. This saves a step.
Yes, I have the same workflow as you.

As an aside, I really don't get the Exercism thing of skipping tests. I really don't think it's that overwhelming to get many failing tests at first. First thing I always do for any exercise is removing the skips. Ah well...

Reply:
> As an aside, I really don't get the Exercism thing of skipping tests.

I'm on your side too. But I remember discussions happening on that subject and skipping was the chosen decision so we comply with the rules ;)